### PR TITLE
FE-966 - Remove references to OG Matchbox styles when Hibana is enabled

### PR DESCRIPTION
**Pull request by nicklemmon**
[FE-966](https://jira.int.messagesystems.com/browse/FE-966)

### What Changed
- Re-introduces `copyMatchboxStyles` script - I know, I've gone back and forth on this one a lot
- Moves OG theme global app CSS to the new `HibanaStyleHandler` component
- CSS from Matchbox renders *before* CSS module CSS dynamically to avoid bugs related to CSS specificity
- Moved critical CSS to a Sass partial

### How To Test

- Verify styles in the OG app remain unchanged
- Verify styles in the Hibana app do not include global styles from the original version of Matchbox

### CODE REVIEW

File: public/index.html [hunk 1]
```diff
@@ -56,22 +56,38 @@
...
+          y.async = !0;
+          y.src = 'https://cdn.pendo.io/agent/static/' + apiKey + '/pendo.js';
+          z = e.getElementsByTagName(n)[0];
+          z.parentNode.insertBefore(y, z);
+        })(window, document, 'script', 'pendo');
       })('be87f509-b8ce-45f4-4c3e-197dcde7ec5a');
-      </script>
+    </script>
+
+    <style type="text/css" id="style-portal"></style>
```
Comment (nicklemmon) [inline public/index.html:82]: This is the only change - the rest is Prettier.

File: src/styles/_critical.scss [hunk 2]
```diff
@@ -47,15 +46,15 @@ html {
 }
 
 @keyframes pulse {
-  0%, 100% {
+  0%,
```
Comment (nicklemmon) [inline src/styles/_critical.scss:49]: The file was just renamed, these other changes are just Prettier.

File: src/index.scss [hunk 3]
```diff
@@ -1,53 +1,3 @@
+@import 'styles/critical';
 @import url('/static/fonts/calibre/calibre.css');
-@import '~@sparkpost/matchbox/styles.scss'; // Global Compiled
 @import '~@sparkpost/matchbox/src/styles/config.scss';
```
Comment (bkemper) [inline src/index.scss:3]: Can't this be removed too?

File: src/styles/_critical.scss [hunk 4]
```diff
@@ -47,15 +46,15 @@ html {
 }
 
 @keyframes pulse {
-  0%, 100% {
```
Comment (bkemper) [inline src/styles/_critical.scss:50]: Moving this fie breaks the path used by Storybook in `.storybook/config.js`.

File: src/components/hibana/HibanaStyleHandler.js [hunk 5]
```diff
@@ -0,0 +1,78 @@
+import ReactDOM from 'react-dom';
+import { useHibana } from 'src/context/HibanaContext';
+
+const baseStyles = `
+@import url(/static/og-matchbox-styles.css);
```
Comment (bkemper) [inline src/components/hibana/HibanaStyleHandler.js:5]: This and the build modifications to copy the css to static can be replaced with this. 

```
import styles from '!!raw-loader!@sparkpost/matchbox/styles.css';

const baseStyles = `
${styles}

html {
  font-size: 16px;
}
```

**Comment by bkemper** [comment]
> There is something off — top is feature branch, bottom is master. Maybe, spacing. I'm looking into it.
> 
> Screen Shot 2020-04-15 at 3 43 51 PM

This is due to font-size.

File: src/index.scss (hunk 3 above)
Comment (nicklemmon) [reply src/index.scss:3]: Let me try...probably!

File: src/components/hibana/HibanaStyleHandler.js (hunk 5 above)
Comment (nicklemmon) [reply src/components/hibana/HibanaStyleHandler.js:5]: Great find. This is awesome.

**Pull request merged**